scylla/src/response/query_result.rs

```diff
@@ -444,6 +458,1180 @@ impl QueryRowsResult {
+        let target_date = base_date
+            + match Duration::try_days(days as i64) {
+                Some(duration) => duration,
+                None => return write!(f, "Invalid date"),
+            };
+
+        // Format the date
+        write_colored!(
+            f,
+            self.settings.print_in_color,
+            self.settings.colors.date,
+            "{}",
+            target_date.format("%Y-%m-%d")
+        )
+    }
+}
+
+// inet
+impl fmt::Display for WrapperDisplay<'_, std::net::IpAddr> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let ip = self.value;
+        match ip {
+            std::net::IpAddr::V4(ipv4) => {
+                write_colored!(
+                    f,
+                    self.settings.print_in_color,
+                    self.settings.colors.inet,
+                    "{}",
+                    ipv4
+                )
+            }
+            std::net::IpAddr::V6(ipv6) => {
+                write_colored!(
+                    f,
+                    self.settings.print_in_color,
+                    self.settings.colors.inet,
+                    "{}",
+                    ipv6
+                )
+            }
+        }
+    }
+}
+
+// boolean
+impl fmt::Display for WrapperDisplay<'_, bool> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write_colored!(
+            f,
+            self.settings.print_in_color,
+            self.settings.colors.boolean,
+            "{}",
+            self.value
+        )
+    }
+}
+
+// Uuid
+impl fmt::Display for WrapperDisplay<'_, Uuid> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write_colored!(
+            f,
+            self.settings.print_in_color,
+            self.settings.colors.uuid,
+            "{}",
+            self.value
+        )
+    }
+}
+
+fn display_in_quotes_and_colored<'a>(
+    f: &mut fmt::Formatter<'_>,
+    value: &(dyn StringConvertible<'a> + 'a),
+    settings: &RowsDisplayerSettings,
+) -> fmt::Result {
+    write_colored!(f, settings.print_in_color, settings.colors.text, "'")?;
+    write_colored!(
+        f,
+        settings.print_in_color,
+        settings.colors.text,
+        "{}",
+        value
+    )?;
+    write_colored!(f, settings.print_in_color, settings.colors.text, "'")
+}
+
+// list or set
+impl fmt::Display for WrapperDisplay<'_, Vec<CqlValue>> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        // choose opening bracket
+        if let CqlValue::List(_) = self.cql_value {
+            write_colored!(
+                f,
+                self.settings.print_in_color,
+                self.settings.colors.collection,
+                "["
+            )?;
+        } else {
+            // it is a set
+            write_colored!(
+                f,
+                self.settings.print_in_color,
+                self.settings.colors.collection,
+                "{{"
+            )?;
+        }
```

Comment:
:wrench: Let's make sure that it is indeed a set.

Reply:
```suggestion
        if matches!(self.cql_value, CqlValue::List(_))  {
            write_colored!(
                f,
                self.settings.print_in_color,
                self.settings.colors.collection,
                "["
            )?;
        } else if matches!(self.cql_value, CqlValue::Set(_)) {
            // it is a set
            write_colored!(
                f,
                self.settings.print_in_color,
                self.settings.colors.collection,
                "{{"
            )?;
        } else {
            panic!("BUG: Misuse of..."); // Describe the reason why it might happen.
        }
```